base/array.jl

```diff
@@ -2294,7 +2294,9 @@ findfirst(A::AbstractArray) = findnext(A, first(keys(A)))
     findnext(predicate::Function, A, i)
 
 Find the next index after or including `i` of an element of `A`
-for which `predicate` returns `true`, or `nothing` if not found.
+for which `predicate` returns `true`, or `nothing` if not found. This works for
+any collection that supports [`getindex`](@ref), [`keys(A)`](@ref), and
+[`nextind`](@ref).
```

Comment:
I don't think it's a good idea to say "this works for anything that X" unless X is a documented interface. For example, `findnext` does not work for the type `OneToInf` defined below even though it meets the requirements posed in this PR's docstring.

```julia
julia> struct OneToInf end

julia> function Base.getindex(oti::OneToInf, i)
           @boundscheck i > 0 || throw(BoundsError(oti, i))
           i
       end

julia> Base.keys(oti::OneToInf) = oti

julia> Base.nextind(::OneToInf, i, n=1) = i+n

julia> findnext(==(7), OneToInf(), 5)
ERROR: MethodError: no method matching iterate(::OneToInf)

Closest candidates are:
  iterate(::Base.AsyncGenerator, ::Base.AsyncGeneratorState)
   @ Base asyncmap.jl:362
  iterate(::Base.AsyncGenerator)
   @ Base asyncmap.jl:362
  iterate(::Pkg.Resolve.NodePerm, Any...)
   @ Pkg ~/.julia/juliaup/julia-1.10.0+0.aarch64.linux.gnu/share/julia/stdlib/v1.10/Pkg/src/Resolve/maxsum.jl:240
  ...

Stacktrace:
 [1] first(itr::OneToInf)
   @ Base ./abstractarray.jl:469
 [2] findnext(testf::Base.Fix2{typeof(==), Int64}, A::OneToInf, start::Int64)
   @ Base ./array.jl:2151
 [3] top-level scope
   @ REPL[84]:1

julia> Base.iterate(::OneToInf, i=0) = (i+1, i+1)

julia> findnext(==(7), OneToInf(), 5)
ERROR: MethodError: no method matching lastindex(::OneToInf)

Closest candidates are:
  lastindex(::Any, ::Any)
   @ Base abstractarray.jl:424
  lastindex(::LaTeXStrings.LaTeXString)
   @ LaTeXStrings ~/.julia/packages/LaTeXStrings/ZtSdh/src/LaTeXStrings.jl:109
  lastindex(::Base.JuliaSyntax.SourceFile)
   @ Base /cache/build/default-armageddon-5/julialang/julia-release-1-dot-10/base/JuliaSyntax/src/source_files.jl:132
  ...

Stacktrace:
 [1] last(a::OneToInf)
   @ Base ./abstractarray.jl:524
 [2] findnext(testf::Base.Fix2{typeof(==), Int64}, A::OneToInf, start::Int64)
   @ Base ./array.jl:2152
 [3] top-level scope
   @ REPL[88]:1
```

I would recommend either
- An affirmative list of things it works for;
- Weaken "any" to "most"; or
- Both: "..Arrays, Strings, and most other collections that support..."

Reply:
Right, good point; changed to the last option in 1a4c54c322.